pkg/policy/repository.go

```diff
@@ -83,9 +84,13 @@ type Repository struct {
 	revision atomic.Uint64
 
 	// selectorCache tracks the selectors used in the policies
-	// resolved from the repository.
+	// resolved from the repository by alive endpoints.
 	selectorCache *SelectorCache
 
+	// policySelectorCache tracks the selectors used by policies
+	// to select what endpoints they apply to
+	policySelectorCache *SelectorCache
```

Comment:
`subjectSelectorCache` might be a more descriptive name for this? It is caching policy subject selectors, after all?